ow_faults/src/FaultInjector.cpp

```diff
@@ -75,16 +76,54 @@ void FaultInjector::setFaultsMessage(ow_faults::PTFaults& msg, ComponentFaults v
   msg.value = static_cast<uint>(value); //should be HARDWARE for now
 }
 
-void FaultInjector::setPowerTemperatureFaultValue(bool getTempBool){
-  float thermal_val;
-  if (isnan(powerTemperatureOverloadValue)) {
-    thermal_val =  50.0 + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(90.0-50.0)));
-    powerTemperatureOverloadValue = thermal_val;
-  } else if (!getTempBool) {
-    powerTemperatureOverloadValue = NAN;
+float FaultInjector::getRandomValueFromRange( float min_val, float max_val){
+  float randomValue = min_val + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(max_val-min_val)));
+  return randomValue;
+}
+
+void FaultInjector::setPowerFaultValues(string powerType, float min_val, float max_val){
```

Comment:
Minor: const string& is better, and idiomatic for pass-by-value strings.

Reply:
good point, will do!